colossalai/pipeline/policy/bert.py

```diff
@@ -254,9 +263,9 @@ def get_hold_layers(self, module: BertModel) -> List[Module]:
         if self.stage_manager.is_first_stage():
             hold_layers.append(module.embeddings)
         num_layers_per_stage_accumulated = np.cumsum(self.layers_per_stage)
-        hold_layers.extend(module.encoder.layer[num_layers_per_stage_accumulated \
-                    [self.stage_manager.stage-1] if self.stage_manager.stage > 0 else 0:
-                    num_layers_per_stage_accumulated[self.stage_manager.stage]])
+        hold_layers.extend(
+            module.encoder.layer[num_layers_per_stage_accumulated[self.stage_manager.stage - 1] if self.stage_manager.
```

Comment:
This is ugly. You can first `np.insert(num_layers_per_stage_accumulated, 0, 0)`, then
```python
start_idx = num_layers_per_stage_accumulated[self.stage_manager.stage - 1]
end_idx = num_layers_per_stage_accumulated[self.stage_manager.stage]
hold_layers.extend(module.encoder.layer[start_idx:end_idx])
```

Reply:
And this process can be extracted to base class.

Reply:
done

Reply:
extracted and also make it a static method